paddle/fluid/imperative/nccl_context.cc

```diff
@@ -206,11 +91,38 @@ void NCCLParallelContext::AllReduceByStream(const framework::Variable &src,
   AllReduce(src, dst, strategy_, ring_id, use_calc_stream);
 }
 
-paddle::platform::CUDADeviceContext *NCCLParallelContext::GetDeviceContext(
+paddle::platform::DeviceContext *NCCLParallelContext::GetDeviceContext(
     int ring_id) {
-  return platform::NCCLCommContext::Instance()
-      .Get(ring_id, place_)
-      ->dev_context();
+  return static_cast<platform::DeviceContext *>(
+      platform::NCCLCommContext::Instance()
+          .Get(ring_id, place_)
+          ->dev_context());
+}
+
+void NCCLParallelContext::WaitCompute(int ring_id) {
+  auto compute_stream = static_cast<platform::CUDADeviceContext *>(
+                            platform::DeviceContextPool::Instance().Get(place_))
+                            ->stream();
+  auto comm_stream =
+      platform::NCCLCommContext::Instance().Get(ring_id, place_)->stream();
+  auto event = compute_events_[ring_id].get();
```

Comment:
add assert for ring_id < compute_events_.size()? Same as WaitComm

Reply:
Done. add assert ring_id >=0 and ring_id < compute_events_.size